### ISSUE TITLE
[BUG] 5.0.0 breaks generator if servers url relative

### ISSUE THREAD

**Issue by UnleashSpirit** [OpenAPITools/openapi-generator#8266]
Hi recently switch from openapi-generator-maven-plugin@4.3.1 to 5.0.0 and the generation of my swagger did not work anymore.

After hours of search, I finaly found that it was mu servers > url value that breaked the generation.
Of course my swagger is valid according to https://apitools.dev/swagger-parser/online/ (and many others).
And also specification : https://swagger.io/specification/#server-object

Field Name | Type | Description
-- | -- | --
url | string | REQUIRED. A URL to the target host. This URL supports Server Variables and **MAY** be relative

### Step to reproduce :
Take the classic petstore : https://raw.githubusercontent.com/OAI/OpenAPI-Specification/master/examples/v3.0/petstore.yaml
Change 
```
servers:
  - url: http://petstore.swagger.io/v1
```
To
```
servers:
  - url: /v1
```

Run
```bash
java -jar openapi-generator-cli.jar generate -i <path_to/petstore.yaml> -g spring
```
Breaks with error (unclear at all)
`com.fasterxml.jackson.core.JsonParseException: Unrecognized token 'openapi': was expecting (JSON String, Number, Array, Object or token 'null', 'true' or 'false')`

I guess it breaks all generator cause it also break validate command.

**Comment by selliera** [comment]
Reverting the version of swagger-parser in 5.0.0 to 2.0.20 enabled me to have a better error message:

```
Exception in thread "main" org.openapitools.codegen.SpecValidationException: There were issues with the specification. The option can be disabled via validateSpec (Maven/Gradle) or --skip-validate-spec (CLI).
 | Error count: 1, Warning count: 0
Errors:
        -attribute .servers.invalid url : /api

        at org.openapitools.codegen.config.CodegenConfigurator.toContext(CodegenConfigurator.java:541)
```

And true enough: adding the --skip-validate-spec to my call lets this work. Ugly, but a suitable workaroud waiting for the problem to be fixed.

**Comment by UnleashSpirit** [comment]
Ho it's not OpenApi so ? It's "deeper".
You know if there is a issue on swagger-parser ?

**Comment by selliera** [comment]
It looks like there is one. In fact when I searched first, I had nto yet understood that it was coming from the url.
https://github.com/swagger-api/swagger-parser/issues/1509 look like the cause of our problems.

**Comment by selliera** [comment]
Oh, my bad: the --skip-validate-spec workaround is only usable with 5.0.0 rebuilt with swagger-parser version 2.0.21 as a dependency. With the real 5.0.0, it does not work at all ! (I tested with the wrong jar)

**Comment by SwingGuy1024** [comment]
I should point out that there are two problems here. One is that a relative url breaks the build. The other is that the error message is incorrect. I'm having a different issue that's breaking the build (https://github.com/SwingGuy1024/OpenAPI_inputSpec_Bug) where I also get a misleading error message, so this may be a wider problem with the code.
I haven't looked at the code, but in my experience, this may happen if exceptions are being handled carelessly. For example, If you catch Exception and assuming it's the specific exception you were expecting, you may produce an incorrect error message if some other bug threw a RuntimeException that's also getting caught. This is why, whenever I need to say "catch Exception", I always also catch RuntimeException and rethrow it. I have no idea if this is the cause of your misleading error messages, but I thought I'd mention it.

**Comment by stapel** [comment]
Might this be a windows-related problem? I had no problems generating a client from a schema with a relative path, a colleague however could not build it (openapi-gen 5.0.1, maven plugin), though he could build it with an absolute path.

**Comment by frochi42** [comment]
Any updates on this? We definitely need that possibility to provide relative URL in servers...
Another question: as I am using the generator from the maven plugin: Is there an example available how to force the usage of a differen swagger parser when using that plugin?

**Comment by deiag** [comment]
I ran into the same issue on Windows, but when I tried it on Ubuntu via WSL, it worked fine, so I assume there's something in the way paths are handled in Windows that's the problem.

**Comment by Klaboe** [comment]
Can chip in here with cliff note of my experience:

Is on Windows10
Has openapi3 spec,
Has relative path i `servers.url`
Used gradle plugin 5.1.0

Validation (and build if one tries to skip validation in the config) breaks and it seems like it has tried to parse the file as en openapi2-spec:

```
Errors:
        -attribute swagger is missing
        -attribute components is unexpected
        -attribute openapi is unexpected
        -attribute servers is unexpected
```

Skipping validation made it generate the api-files, but not the component-files.

If i tried building from WSL, everything worked out fine
If i tried generation with the CLI for 5.1.0, everything worked out fine

**Comment by jhannes** [comment]
I think this is the underlying cause: https://github.com/swagger-api/swagger-parser/issues/1553

**Comment by norrs** [comment]
What is the usual timeframe for making a new release of `openapi-generator` in maven repositories which includes the fix shipped in https://github.com/swagger-api/swagger-parser/issues/1553? 
Is this considered critical enough to warant a quick release for the team? (cc @wing328 as I see the user in the Github releases page) 

As currently this breaks builds for our Windows developers without involving and running the maven target for openapi-generator under WSL2 in Windows after we upgraded from 4.3.1 to 5.1.0.

Thanks @jhannes for finding the underlaying cause and fixing this upstream.

**Comment by lufecir** [comment]
Yes, same question as norrs. Would also much appreciate a release.
 So any idea on when we might expect this? To understand what our options are.

**Comment by jhannes** [comment]
Please note that swagger-parser hasn't been released with the PR yet and I have not verified that it does in fact fix the problem in openapi-generator. So the request for a new release must start upstream :-)

**Comment by jhannes** [comment]
Swagger-parser 2.0.26 was just released with this fix. Upgrading the dependency in openapi-generator should allow it to work on Windows again

**Comment by marcelstoer** [comment]
This is a duplicate of #8260.
